### PR TITLE
Add live collaboration between users

### PR DESCRIPTION
**Pull request by tyouwei**
To test locally:
1. Run docker-compose up --build
2. Open an incognito browser and a normal browser
3. Login to different accounts on both browsers
4. Match both users using matching feature
5. Type in the code editor and observe

O ya rmb to get the updated env from our drive

### CODE REVIEW

File: matching-service/src/lib/hasher.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import crypto from 'crypto';
+
+export const generateMatchId = (userId: string, matchId: string): string => {
```
Comment (Kurtyjlee) [inline matching-service/src/lib/hasher.ts:3]: I think the indentation here is 4 spaces instead of 2, can run prettier to fix this

Comment (Kurtyjlee) [reply matching-service/src/lib/hasher.ts:3]: Nice function tho

File: peerprep-fe/src/lib/socket.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// pages/api/socket.js
+
+import { WebSocketServer } from 'ws';
+
+export default function handler(req: any, res: any) {
```
Comment (Kurtyjlee) [inline peerprep-fe/src/lib/socket.ts:5]: Here as well, the styles a bit off

**Review by Kurtyjlee — COMMENTED** [review]
Nice feature bro, the coding component is super responsive. I only got comments on style. Try to do build using the docker compose prod file, it will surface any build/linter issues

File: peerprep-fe/src/lib/socket.ts (hunk 2 above)
Comment (tyouwei) [reply peerprep-fe/src/lib/socket.ts]: hmm, my bad this entire file isnt supposed to be here LOL. removign this

File: matching-service/src/lib/hasher.ts (hunk 1 above)
Comment (tyouwei) [reply matching-service/src/lib/hasher.ts:3]: fixed. One a sidenote, how are you guys seeing this linting issues? I cannot see them on my side.

Comment (Kurtyjlee) [reply matching-service/src/lib/hasher.ts:3]: Have you set up husky?

**Review by lsyurea — APPROVED** [review]
LGTM, I have tested it and it is working as expected
Minor nits, I think in the .gitignore file, & should not have appeared

**Pull request merged**